openmc/lattice.py

```diff
@@ -759,6 +770,277 @@ def is_valid_index(self, idx):
+                    if strategy == "degenerate":
+                        break
+
+                    # Look at all rotations of pattern
+                    for rot in range(4):
+                        if not found and tuple(map(tuple, pattern)) == known_pattern:
+                            found = True
+
+                            # Save location of the pattern in the lattice
+                            pattern_data['locations'].append((i, j))
+
+                        # Rotate pattern
+                        pattern = np.rot90(pattern)
+
+                    # Look at transpose of pattern and its rotations
+                    pattern = np.transpose(pattern)
+                    for rot in range(4):
+                        if not found and \
+                             tuple(map(tuple, pattern)) == known_pattern:
+                            found = True
+
+                            # Save location of the pattern in the lattice
+                            pattern_data['locations'].append((i, j))
+
+                        # Rotate pattern
+                        pattern = np.rot90(pattern)
+
+                    # Transpose pattern back for the next search
+                    pattern = np.transpose(pattern)
+
+                # Create new pattern and add to the patterns dictionary
+                if not found:
+                    pattern_data = {}
+                    pattern_data['locations'] = [(i, j)]
+                    if strategy == "lns":
+                        patterns[tuple(map(tuple, pattern))] = pattern_data
+                    elif strategy == "degenerate":
+                        patterns[(i,j)] = pattern_data
+
+        # Discretize lattice
+        for pattern, pattern_data in patterns.items():
+
+            first_pos = pattern_data['locations'][0]
+
+            # Create a clone of the universe, without cloning materials
+            new_universe = self.universes[first_pos[1]][first_pos[0]].clone(
+                 clone_materials=False, clone_regions=False)
+
+            # Discretize the immediate sub-lattices of this universe
+            for sub_cell in new_universe.cells.values():
+
+                sub_universe = sub_cell.fill
+
+                try:
+                    sub_universe.discretize(strategy,
+                                            universes_to_ignore,
+                                            materials_to_clone,
+                                            lattice_neighbors=[],
+                                            attribute=attribute)
+                except AttributeError:
+                    continue
+
+            # Replace only the materials in materials_to_clone
+            for material in materials_to_clone:
+                material_cloned = False
+
+                for cell in new_universe.get_all_cells().values():
+
+                    if cell.fill is not None:
+                        if cell.fill.id == material.id:
```

Comment:
The cell fill could be a `Universe` so you probably need to check that this fill is a `Material`.

Also, there will be an exception here for cells with distribmats. I'm not really sure how LNS should handle distribmats so you should probably just raise a `ValueError` if a distribmat is detected in the lattice.

Reply:
Added. Also added an error for distribcells, detected through the length of their temperature and fill attributes.